notes and questions n27460 - Add importmempool RPC (#667)
bitcoin-core-review-club/website: _posts/2023-04-26-#27460.md

components: ["rpc/rest/zmq"]
host: larryruane
status: upcoming
commit: fab8b370257c7770abc32649ad4940eefc512f44
---

## Notes

- The [mempool](https://bitcoin.stackexchange.com/questions/tagged/mempool)
  is the list of unconfirmed (pending) transactions.
  (_Rabbit hole warning_) More information on the mempool can be found
  [here](https://bitcoinsearch.xyz/?q=mempool)

- Initially, the mempool was stored only in memory, as its name implies.
  [PR 8448](https://github.com/bitcoin/bitcoin/pull/8448)
  implemented persisting the mempool to disk so that its entries are
  available after a restart. This PR was merged in
  [v0.14.0](https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-0.14.0.md#retaining-the-mempool-across-restarts).

- The `mempool.dat` file, located in the datadir, is a binary file in a
  proprietary format, so it's not possible
  to manually edit this file.

> - The in-memory mempool is not just a cache;
>   it contains the entire mempool rather than a subset (such as most-recently-used).

Reviewer: Would slightly rephrase, e.g. like this?
```suggestion
- The entire mempool is kept in-memory, it is not just a cached subset of a larger structure.
```
